packages/math-input/src/components/keypad-button.tsx

```diff
@@ -257,7 +257,7 @@ class KeypadButton extends React.PureComponent<Props> {
                 <View style={buttonStyle} {...eventHandlers} {...a11yMarkup}>
                     {maybeFocusBox}
                     <View style={iconWrapperStyle}>
-                        <Icon icon={icon} focused={renderFocused} />
+                        <Icon icon={icon as IconType} focused={renderFocused} />
```

Comment:
So a lot of the forced typing and optional types seem odd here. My understanding is that it depends on what kind of button we're rendering.

For instance, if we're rendering a button that is a `MANY` button, it doesn't really have its own icon - so `icon` is optional. But at certain point I had to say "I'm pretty sure we'll have XYZ here" so I used `as` a couple of times.

Reply:
Technically, all key configs have a configured icon except for maybe `MANY` (it looks like we manually create a `MANY` keyconfig in places. 

I have ideas for how we might make these types cover what we're doing... Something like this: 

```

export type KeyConfig =
    | {
          id: Omit<Key, "MANY">;
          type: Omit<keyof typeof KeyTypes, "MANY">;
          icon: Icon;

          childKeyIds?: ReadonlyArray<string>;
          ariaLabel?: string;
      }
    | {
          id: "MANY";
          type: "MANY";
          childKeyIds: ReadonlyArray<string>;
          ariaLabel?: string;
      };
```

Happy to pair tomorrow (Apr 20) with you if you want to try to get this working. :)